src/oak.py

```diff
@@ -540,6 +544,26 @@ async def get_properties(
         """
         return self.camera_properties
 
+    def _wait_until_worker_running(self, max_attempts=5, sleep_time=1):
+        """
+        Blocks execution until worker is running.
+
+        Args:
+            max_attempts (int, optional): Defaults to 5.
+            sleep_time (int, optional): Defaults to 1.
+
+        Raises:
+
+        """
+        cls: Oak = type(self)
+        attempts = 0
+        while attempts < max_attempts:
+            if cls.worker.running:
+                return
+            time.sleep(sleep_time)
+            attempts += 1
+        raise ViamError("camera data requested before camera worker was ready.")
```

Comment:
add more user facing instructions to the error, what do i do here?

Reply:
I think the new message suffices. LMK ur thoughts